src/System.IO.Compression/tests/DeflateStreamTests.cs

```diff
@@ -1,23 +1,40 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
+using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
-using System.IO;
-using System.Reflection;
-using System.IO.Compression;
 using Xunit;
 
 namespace System.IO.Compression.Tests
 {
-    public partial class DeflateStreamTests
+    public class ZLibDeflateStreamTests : DeflateStreamTests, IDisposable
+    {
+        public ZLibDeflateStreamTests() { SetWorkerMode("zlib"); }
+        public void Dispose() { SetWorkerMode("unknown"); }
+    }
+
+    public class ManagedDeflateStreamTests : DeflateStreamTests, IDisposable
+    {
+        public ManagedDeflateStreamTests() { SetWorkerMode("managed"); }
+        public void Dispose() { SetWorkerMode("unknown"); }
+    }
+
+    public abstract class DeflateStreamTests
     {
         static string gzTestFile(String fileName) { return Path.Combine("GZTestData", fileName); }
 
+        internal static void SetWorkerMode(string mode)
+        {
+            FieldInfo forceType = typeof(DeflateStream).GetTypeInfo().GetDeclaredField("s_forcedTestingDeflaterType");
+            if (forceType != null)
+            {
+                forceType.SetValue(null, mode == "zlib" ? (byte)2 : mode == "managed" ? (byte)1 : (byte)0);
+            }
```

Comment:
Else (eg. running these tests on ret) put a warning on the console, or fail the test?